ecs-agent/netlib/platform/common_linux.go

```diff
@@ -120,6 +125,14 @@ func NewPlatform(
 				common: commonPlatform,
 			},
 		}, nil
+	case ManagedPlatform:
+		config := aws.NewConfig().WithMaxRetries(metadataRetries)
+		config.Credentials = instancecreds.GetCredentials(false)
+		ec2Client := ec2metadata.New(session.New(), config)
```

Comment:
Does this call always succeed?

Reply:
This is the same call that the top level client would do in initialization there technically is the possibility of failure which is why we have 5 retries